Sources/MapboxNavigation/NavigationViewController.swift

```diff
@@ -324,6 +330,23 @@ open class NavigationViewController: UIViewController, NavigationStatusPresenter
         }
     }
     
+    func setupPredictiveCaching(_ navigationOptions: NavigationOptions?) {
```

Comment:
This is code that a developer would have to rewrite if they’re implementing their own turn-by-turn navigation view controller based on NavigationMapView and RouteController. We’ll also have to rewrite it for CarPlayNavigationViewController. Should this code be pushed into NavigationMapView?

Reply:
I agree that predictive cache functionality should be part of `NavigationMapView`, this will also allow us to control `PredictiveCacheOptions` via `NavigationMapView` interface, instead of extending `NavigationOptions`.

Reply:
I think you are right, it makes sense to have this in `NavigationMapView` instead. But I still think that we should have `PredictiveCacheOptions` inside `NavigationOptions` since this is our default way for customising navigation experience when used together with `NavigationViewController`. At the same time it'll be still possible to configure Predictive Caching from `NavigationMapView` directly.